alchemy/src/coinbase/faucet.ts

```diff
@@ -0,0 +1,183 @@
+        );
+        for (const s of possibleScopes) {
+          console.log(`  - ${s}`);
+        }
+
+        // Collect state files from all matching scopes
+        stateFiles = [];
+        for (const foundScope of possibleScopes) {
+          const files = await glob("*.json", {
+            cwd: join(".alchemy", foundScope),
+            absolute: true,
+          });
+          stateFiles.push(...files);
+        }
+      }
+    }
+  } else {
+    // No scope provided, search all .alchemy subdirectories
+    console.log(
+      "Searching all scopes for accounts with faucet configuration...",
+    );
+    stateFiles = await glob("**/*.json", {
+      cwd: ".alchemy",
+      absolute: true,
+    });
+  }
+
+  const allStates = (
+    await Promise.all(
+      stateFiles.map(async (file) => {
+        try {
+          const content = await readFile(file, "utf-8");
+          return JSON.parse(content);
+        } catch {
+          return null;
+        }
+      }),
+    )
+  ).filter(Boolean);
+
+  const accounts = allStates
+    .filter(
+      (state) =>
+        state?.kind?.startsWith("coinbase::evm") &&
+        state.output &&
+        (state.output as EvmAccount | EvmSmartAccount).faucet,
+    )
+    .map((state) => {
+      const output = state.output as EvmAccount | EvmSmartAccount;
+      return {
+        address: output.address,
+        name: output.name,
+        faucet: output.faucet!,
+      };
+    });
+
+  if (accounts.length === 0) {
+    console.log("No accounts with faucet configuration found");
+    return;
+  }
+
+  console.log(`Found ${accounts.length} accounts with faucet configuration\n`);
+
+  for (const account of accounts) {
+    console.log(`Account: ${account.name} (${account.address})`);
+
+    for (const [network, tokens] of Object.entries(account.faucet)) {
+      for (const token of tokens) {
+        const key = `${account.address}-${network}-${token}`;
+
+        // Skip if already processed in this run
+        if (funded.has(key)) {
+          console.log(
+            `  ⏭️  Skipping ${token} on ${network} (already requested)`,
+          );
+          continue;
+        }
+
+        // Request tokens from faucet
+        try {
+          console.log(`  💧 Requesting ${token} on ${network}...`);
+
+          const response = await cdp.evm.requestFaucet({
+            address: account.address,
+            network: network as FaucetNetwork,
+            token: token as FaucetToken,
+          });
+
+          console.log(`  ✅ Funded: ${response.transactionHash}`);
+          funded.add(key);
+
+          // Small delay to avoid rate limits
+          await new Promise((resolve) => setTimeout(resolve, 1000));
+        } catch (error: any) {
+          console.log(
+            `  ❌ Error funding ${token} on ${network}: ${error.message}`,
+          );
+        }
+      }
+    }
+    console.log();
+  }
+
+  console.log("✨ Funding complete!");
+}
+
+main().catch(console.error);
```

Comment:
No need for the main function. Just use top-level await

Reply:
addressed in https://github.com/alchemy-run/alchemy/pull/1012/commits/d0fdfcd418531cd49a841adb6f4fffc92df390bc